ccm_web/server/src/api/api.utils.ts

```diff
@@ -63,3 +64,19 @@ export function makeResponse<T> (multipleResults: Array<APIErrorData | T>): T[]
     }
   }
 }
+
+/*
+Convenience wrapper for p-limit: see https://github.com/sindresorhus/p-limit
+*/
+export function createLimitedPromises<T> (
+  funcs: Array<() => Promise<T>>, maxConcurrentPromises = 20
```

Comment:
I just changed the default here from `5` to `20`. I've been trying a few different values, and observing how the `x-rate-limit-remaining` value changes. I tried `10`, `20`, and `40`. With some repeated batches of 400, `40` was triggering some retries. `20` still seemed to be super fast, with runs taking 12, 13, 18 seconds, and I never saw retries, even with batches in quick succession. The `x-rate-limit-remaining` seems to have a maximum of `700`, and I never saw it go below `400`. It certainly needs more testing, but this feels pretty good to me. If the results are this consistent, we may want to change the loading message so it reflects the typical waiting time.

Reply:
can the `maxConcurrentPromises` be a configurable option?

Reply:
We certainly coould, but in what situation can you imagine where we would want to be able to change this in production? I feel like it's better to just test to make sure it performs well with a certain number. I'm just worried about making the configuration too complicated (an issue we've had with other projects).

Reply:
We got too many configuration value, I am ok to leave it. Since if this works consistently we never going to change it and I am not anticipating we will be supporting 700 users enrollment.

Reply:
I guess it's possible Canvas could adjust their rate limit, and we may need to respond to that. Hm... it's possible to make it configurable, but as it is now, we'd have to pass in a configuration variable all the time (since this function lives in `api.utils`, and I can't simply inject `ConfigService` here. Maybe I can rework this so that isn't necessary...

Reply:
@ssciolla Sorry for not commenting before. I think it is ok not make the concurrency configurable. We can think about it after the initial release

Reply:
Okay, we can wait on this.